client/lib/after-layout-flush/index.ts

```diff
@@ -12,18 +14,23 @@
  * https://developer.mozilla.org/en-US/Firefox/Performance_best_practices_for_Firefox_fe_engineers
  *
  * @param {Function} func The function to be invoked after the layout flush
- * @returns {Function} Returns the new delayed function
+ * @returns {Function} The new delayed function
```

Comment:
I've been removing the types from the JSDoc comments when adding them in the code. To me it makes sense not to keep two copies of that information next to each other when one will be tied to the code and the other manually updated. Already we have a divergence here because the JSDoc doesn't mention anything about being cancelable.

Reply:
I adctually did that at first, but it currently produces linting errors with our config:

```
<path>/wp-calypso/client/lib/after-layout-flush/index.ts
   5:1  error  Missing JSDoc return type                valid-jsdoc
  16:4  error  Missing JSDoc parameter type for 'func'  valid-jsdoc
```

Should we remove JSDoc linting from TS files if we expect types to be omitted in the doc (which makes a lot of sense)?

Reply:
that's odd. I thought I removed those lint lines. In #32872 I added `requireParamType: false` to our `eslint` config…

https://github.com/Automattic/wp-calypso/blob/master/packages/eslint-config-wpcalypso/index.js#L113

Reply:
I suppose I could remove the JSDoc altogether and convert this to a standard comment, but that seems less than ideal, as I suspect IDEs won't pick it up.

Reply:
Actually, I'll add a comment for now, to disable the `valid-jsdoc` rule throughout the file.

Reply:
feel free to add that property to the `eslint` rule - I'm pretty sure we'll want it generally going forward, or create a new PR to add it, or wait and I'll try to make a PR to add it